.github/workflows/testing.yml

```
@@ -30,19 +30,39 @@ jobs:
             coverage run --omit=*/test_* -m unittest discover
 
         - name: Create Coverage Report
+          if: steps.check_tests.outputs.has_tests == 'true'
           run: |
             coverage report -m
             coverage xml
 
         - name: DeepSource Test Coverage
+          if: steps.check_tests.outputs.has_tests == 'true'
           uses: deepsourcelabs/test-coverage-action@master
           with: 
```

yamllint (1.35.1)

[error] trailing spaces

(trailing-spaces)